pkg/proto/transactions_test.go

```diff
@@ -6927,3 +6929,222 @@ func TestShadowedCreateAliasWithSig_DoesNotImplementJSONMarshaler(t *testing.T)
+	for i, tst := range []struct {
+		schema byte
+		ver    byte
+		seed   string
+		start  uint32
+		fee    uint64
+		ts     uint64
+	}{
+		{schema: 'W', ver: 1, seed: "COMMIT_TO_GENERATION_TEST_SEED_1", start: 100, fee: 100_000, ts: 1630000000000},
+		{schema: 'T', ver: 1, seed: "COMMIT_TO_GENERATION_TEST_SEED_2", start: 200, fee: 1_0000_0000, ts: 1640000000000},
+	} {
+		t.Run(fmt.Sprintf("%d", i+1), func(t *testing.T) {
+			sk, pk, err := crypto.GenerateKeyPair([]byte(tst.seed))
+			require.NoError(t, err)
+			blsSK, err := bls.GenerateSecretKey([]byte(tst.seed))
+			require.NoError(t, err)
+			blsPK, err := blsSK.PublicKey()
+			require.NoError(t, err)
+			_, sig, err := bls.ProvePoP(blsSK, blsPK, tst.start)
+			require.NoError(t, err)
+			tx := NewUnsignedCommitToGenerationWithProofs(tst.ver, pk, tst.start, blsPK, sig, tst.fee, tst.ts)
+			err = tx.GenerateID(tst.schema)
+			require.NoError(t, err)
+
+			b, err := tx.MarshalToProtobuf(tst.schema)
+			require.NoError(t, err)
+			var atx CommitToGenerationWithProofs
+			err = atx.UnmarshalFromProtobuf(b)
+			require.NoError(t, err)
+			assert.Equal(t, *tx, atx)
+
+			err = tx.Sign(tst.schema, sk)
+			require.NoError(t, err)
+
+			ok, err := tx.Verify(tst.schema, pk)
+			require.NoError(t, err)
+			assert.True(t, ok)
+
+			b, err = tx.MarshalSignedToProtobuf(tst.schema)
+			require.NoError(t, err)
+			var stx CommitToGenerationWithProofs
+			err = stx.UnmarshalSignedFromProtobuf(b)
+			require.NoError(t, err)
+			err = stx.GenerateID(tst.schema)
+			require.NoError(t, err)
+			assert.Equal(t, *tx, stx)
+		})
+	}
+}
+
+func TestCommitToGenerationWithProofsToJSON(t *testing.T) {
+	for i, tst := range []struct {
+		schema byte
+		ver    byte
+		seed   string
+		start  uint32
+		fee    uint64
+		ts     uint64
+	}{
+		{schema: 'W', ver: 1, seed: "COMMIT_TO_GENERATION_TEST_SEED_1", start: 100, fee: 100_000, ts: 1630000000000},
+		{schema: 'T', ver: 1, seed: "COMMIT_TO_GENERATION_TEST_SEED_2", start: 200, fee: 1_0000_0000, ts: 1640000000000},
+	} {
+		t.Run(fmt.Sprintf("%d", i+1), func(t *testing.T) {
+			sk, pk, err := crypto.GenerateKeyPair([]byte(tst.seed))
+			require.NoError(t, err)
+			blsSK, err := bls.GenerateSecretKey([]byte(tst.seed))
+			require.NoError(t, err)
+			blsPK, err := blsSK.PublicKey()
+			require.NoError(t, err)
+			_, sig, err := bls.ProvePoP(blsSK, blsPK, tst.start)
+			require.NoError(t, err)
+
+			tx := NewUnsignedCommitToGenerationWithProofs(tst.ver, pk, tst.start, blsPK, sig, tst.fee, tst.ts)
+
+			js, err := json.Marshal(tx)
+			require.NoError(t, err)
+			ej := fmt.Sprintf(
+				"{\"type\":19,\"version\":%d,\"senderPublicKey\":\"%s\",\"fee\":%d,\"timestamp\":%d,"+
+					"\"generationPeriodStart\":%d,\"endorserPublicKey\":\"%s\",\"commitmentSignature\":\"%s\"}",
+				tst.ver, base58.Encode(pk.Bytes()), tst.fee, tst.ts, tst.start, base58.Encode(blsPK.Bytes()),
+				base58.Encode(sig.Bytes()),
+			)
+			require.Equal(t, ej, string(js))
```

Comment:
LGTM, but suggest using `JSONEq` instead of strict `Equal` here and below.

Reply:
Did as suggested